### PR TITLE
Add test for cloudkitty charm

### PR DESCRIPTION
**Pull request by hernandanielg**
## Add test for cloudkitty charm

For this test we created a class named `CloudkittyTest` 

- Interacts with cloudkitty to test API

### CODE REVIEW

**Comment by camille-rodriguez** [comment]
@hernandanielg pep8 tests failed, and this review is still marked as draft

File: zaza/openstack/charm_tests/cloudkitty/tests.py [hunk 1]
```diff
@@ -44,53 +43,86 @@ def setUpClass(cls):
...
         field = hashmap.create_field(name='test', service_id=service_id)
 
-        field_id = field['field_id']
+        field_id = field.get('field_id')
         mapping = hashmap.create_mapping(
             type='flat', field_id=field_id, value='test', cost=0.1)
 
-        mapping_id = mapping['mapping_id']
+        mapping_id = mapping.get('mapping_id')
         hashmap.delete_mapping(mapping_id=mapping_id)
```
Comment (wolsen) [inline zaza/openstack/charm_tests/cloudkitty/tests.py:113]: Hmm, this is all cleanup stuff right? This is generally done within the test case itself to do the cleanup - which is handled even in the event of certain errors, etc. We should push the deletes/cleanups to the cleanup functions.

File: zaza/openstack/charm_tests/cloudkitty/tests.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+            session=cls.keystone_session
+        )
+
+    def test_400_api_connection(self):
+        """Simple api calls to check service is up and responding."""
+        report = self.cloudkitty.report
+        tenants_list = report.get_tenants()
+        assert tenants_list == []
+
+    def test_401_module_enable_and_disable(self):
```
Comment (camille-rodriguez) [inline zaza/openstack/charm_tests/cloudkitty/tests.py:50]: We should be testing all modules (ideally), or name this test test_401_module_hashmap_enable_and_disable

File: zaza/openstack/charm_tests/cloudkitty/tests.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        service = hashmap.create_service(name='test')
+
+        service_id = service.get('service_id')
+        field = hashmap.create_field(name='test', service_id=service_id)
+
+        field_id = field.get('field_id')
+        hashmap.delete_field(field_id=field_id)
+        hashmap.delete_service(service_id=service_id)
+
+    def test_404_mapping_create_and_delete(self):
```
Comment (camille-rodriguez) [inline zaza/openstack/charm_tests/cloudkitty/tests.py:97]: This test actually includes test 401,402 and 403. They are not needed if we are testing those components in the test here, so I would keep only test 404.

File: zaza/openstack/charm_tests/cloudkitty/tests.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        """Test mapping create and delete via API."""
+        rating = self.cloudkitty.rating
+        rating.update_module(module_id='hashmap', enabled=True)
+
+        hashmap = self.cloudkitty.rating.hashmap
+        service = hashmap.create_service(name='test')
```
Comment (camille-rodriguez) [inline zaza/openstack/charm_tests/cloudkitty/tests.py:103]: The service, mapping and field are all named "test". Please use more specific naming, i.e test-service, test-module, test-field.

File: zaza/openstack/charm_tests/cloudkitty/tests.py (hunk 3 above)
Comment (camille-rodriguez) [inline zaza/openstack/charm_tests/cloudkitty/tests.py:97]: This test actually includes test 401,402 and 403. They are not needed if we are testing those components in the test here, so I would keep only test 404.

File: zaza/openstack/charm_tests/cloudkitty/tests.py [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+    def test_405_group_create_and_delete(self):
+        """Test group create and delete via API."""
+        rating = self.cloudkitty.rating
+        rating.update_module(module_id='hashmap', enabled=True)
+
+        hashmap = self.cloudkitty.rating.hashmap
+        group = hashmap.create_group(name='test')
+
+        assert group.get('name') == 'test'
```
Comment (camille-rodriguez) [inline zaza/openstack/charm_tests/cloudkitty/tests.py:125]: test-group or other more specific naming please

File: zaza/openstack/charm_tests/cloudkitty/tests.py (hunk 1 above)
Comment (hernandanielg) [reply zaza/openstack/charm_tests/cloudkitty/tests.py]: created a teardown function to handle this ;)

File: zaza/openstack/charm_tests/cloudkitty/tests.py (hunk 2 above)
Comment (hernandanielg) [reply zaza/openstack/charm_tests/cloudkitty/tests.py:75]: done +1

**Comment by hernandanielg** [comment]
@camille-rodriguez thanks for your review - I have made changes to the code according to your suggestions

**Pull request merged**